### DISCUSSION TITLE
How do I share dependencies between solutions

### DISCUSSION THREAD

**Discussion by rg-software** [microsoft/vcpkg#19321] [Q&A]
It is recommended to use manifest mode by default, so I am trying to play with it. 

Suppose I develop a static library `MyLib` that uses Boost.  I define `"boost"` as its dependency in the manifest. Then I have a project `MyProj` that uses `MyLib`. I include its header like `#include <MyLib/myheader.h>` and immediately get a compilation error because `myheader.h` happens to use Boost, and of course `MyProj` has no idea about it. Furthermore, if I list `boost` as `MyProj`'s dependency, it might end up getting a different version of Boost, which doesn't sound nice.

Thus, it seems that there is a need to be able to share manifests across solutions of I am doing something wrong. Would much appreciate any advice.

**Comment by autoantwort** [comment]
Is MyProj in the same solution as MyLib? We use a single manifest for all projects inside a solution. If you use vcpkg to depend on MyLib you can add boost as a dependency in the port files vcpkg.json

**Comment by rg-software** [comment]
No, unfortunately. I know I can share the same manifest within the same `sln`, but the issue arises when I create a library that is later used in several other solutions.

**Comment by Neumann-A** [comment]
MSBuild does not have any kind of external dependency lookup. Either you use vcpkg in the whole tree or you just start and use CMake which takes care of this for you.

**Comment by autoantwort** [comment]
Are the solutions in the same git? And how do you distribute MyLib so that you can use the lib in other solutions?

**Comment by rg-software** [comment]
@Neumann-A, Right, but the question is what is "tree". In case of vcpkg it seemingly means "everything under the same `sln` file", which is quite restrictive.

@autoantwort, I am flexible in my setup, but currently what I do is this. I have some homemade libraries I use in several projects. I keep them under `%MY_LIBS%` path. When I want to use a library, I simply add `%MY_LIBS%` to the include and library search paths of a project.

**Reply by Osyotr** [reply]
Take a look at [registries](https://github.com/microsoft/vcpkg/blob/master/docs/users/registries.md).

I think this is somewhat related: https://github.com/microsoft/vcpkg/discussions/18122

**Reply by rg-software** [reply]
Thank you, will take a look!
